src/nvim/shada.c

```diff
@@ -1608,11 +1622,11 @@ static char *shada_filename(const char *file)
           if (STRCMP("$VIM", NameBuff) != 0) {  // $VIM was expanded
             file = SHADA_FILE2;
           } else {
-            file = SHADA_FILE;
+            file = shada_get_default_file();
           }
         } else {
 #endif
-          file =  SHADA_FILE;
+          file =  shada_get_default_file();
 #ifdef SHADA_FILE2
```

Comment:
I have no idea. Inherited from viminfo. Maybe for windows `.`/`_` support, in which case it is not needed (no leading dots in file name anymore).